ospd/server.py

```diff
@@ -67,12 +71,16 @@ def write(self, data: bytes):
 
         while True:
             if b_end > len(data):
-                self.socket.send(data[b_start:])
+                try:
+                    self.socket.send(data[b_start:])
+                except (socket.error, BrokenPipeError) as e:
+                    logger.error("Error sending data to the client. %s", e)
+                    return
```

Comment:
Not sure if the indentation is correct in this sugesstion
```suggestion
                finally:
                    return
```